crates/aranya-fast-channels/src/shm/shared.rs

```diff
@@ -1047,10 +1054,15 @@ impl<CS: CipherSuite> ChanListData<CS> {
         } else {
             // No need to perform a swap if there is only one
             // channel.
-            if len > 1 {
-                self.chans_mut()?.swap(idx, len - 1);
+            if let Some(swap_idx @ 1..) = len.checked_sub(1) {
+                self.chans_mut()?.swap(idx, swap_idx);
+            }
+            // Use the existing arithmetic infrastructure which is designed
+            // to handle wraparound behavior safely.
+            #[allow(clippy::arithmetic_side_effects)] // U64::SubAssign handles this safely
+            {
+                self.len -= 1;
```

Comment:
Use `self.len = self.len.wrapping_sub(1)` and add a comment that it cannot wrap because we already checked the case where `len == 0`.

Reply:
Note that this means you probably need to update `U64` with `checked_sub`, `checked_add`, etc. methods, then use those here and elsewhere.